modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -1335,7 +1357,7 @@ def _index_name(self, col):
         match = re.search("__index__\\d+_(.*)", col)
         if match:
             name = match.group(1)
-            if name == "__None__":
+            if name in ["__None__", "__reduced__"]:
```

Comment:
Could you please explain the case when the index name is "__reduced__"? I had an impression that "__reduced__" is used for an index value, not a name (at least that's what is checked in the QueryCompiler).

Reply:
Imagine we're grouping on Series object, its query compiler has `columns = ["__reduced__"]`, so technically we're doing `df.groupby(by="__reduced__")`. In the result of groupby all by-column became an index, so the resulting index has `__reduced__` **name** (technically it means `None` name). That case presented in [`test_groupby_series`](https://github.com/modin-project/modin/pull/2865/files#diff-207cd19bf5b9d82183530af13df79e0f0d0e870b8ca91f0dc16d1c2a4ed66b03R692).

There is also a different case that you mentioned when index contains `__reduced__` **value**, which means that this axis is reduced and frame has to be squeezed on that axis at the API layer (`axis=1` - frame is unnamed Series, `axis=0` - frame is transposed Series or scalar). That case does not correlate with this change.